service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -551,6 +566,14 @@ public void nameSiteEntry(
     siteEntry.store();
   }
 
+  private SiteEntriesRecord latestSiteEntry(int siteId) {
+    return db.selectFrom(SITE_ENTRIES)
+        .where(SITE_ENTRIES.SITE_ID.eq(siteId))
+        .orderBy(SITE_ENTRIES.CREATED_AT.desc())
+        .fetchInto(SiteEntriesRecord.class)
+        .get(0);
```

Comment:
⛏️ Right now this method assumes there's at least one entry for the site, which might not always be the case - calling this method with a site ID with no entry will throw an `IndexOutOfBoundsException`. Can we add a check to make sure we fetched at least one site entry?

Reply:
It's p counterintuitive but all sites actually (or should at least) have at least one site entry, regardless of whether a tree has ever been planted there or not (at the very least, a site where a tree has never been planted should have a single "site entry" with `treePresent` of `false`, like [this](https://map.treeboston.org/tree/3311014) site). And if a site has zero site entries, then I think it's a data quality issue and we should just add an "empty" site entry for that site. What's funny is that as of now, you can actually adopt, record a stewardship activity for, name, etc. an open planting site lol. Maybe we should take care of this in a separate PR

Reply:
Ah gotcha, thanks for clearing that up - yeah I feel like we shouldn't allow adopting open sites, since there's nothing to actually adopt there yet